src/components/List/ListItem/ListItem.jsx

```diff
@@ -165,14 +156,51 @@ const ListItem = ({
 
   const renderTags = () => (tags && tags.length > 0 ? <div>{tags}</div> : null);
 
+  const renderDragPreview = () => {
+    if (editingStyle && connectDragPreview) {
+      return connectDragPreview(
+        <div className={`${iotPrefix}--list-item-editable--drag-preview`}>
+          {dragPreviewText || value}
+        </div>
+      );
+    }
+
+    return null;
+  };
+
+  const dragIcon = () =>
+    editingStyle ? (
+      <Draggable16
+        className={classnames(`${iotPrefix}--list-item--handle`)}
+        data-testid="list-item-editable"
+      />
+    ) : null;
+
   return (
-    <ListItemWrapper {...{ id, isSelectable, selected, isLargeRow, onSelect }}>
+    <ListItemWrapper
+      isPreview={false}
+      {...{
+        id,
+        isSelectable,
+        selected,
+        isDragging,
+        editingStyle,
+        expanded,
+        isLargeRow,
+        onSelect,
+        connectDragSource,
+        onItemMoved,
+        itemWillMove,
+      }}
+    >
+      {renderDragPreview()}
+      {dragIcon()}
       {renderNestingOffset()}
       {renderExpander()}
       <div
         className={classnames(
           `${iotPrefix}--list-item--content`,
-          { [`${iotPrefix}--list-item--content__selected`]: selected },
+          { [`${iotPrefix}--list-item--content__selected`]: editingStyle ? null : selected },
```

Comment:
Is there a reason for the condition and the null value? Otherwise we could just write !editingStyle && selected